trieste/acquisition/multi_objective/partition.py

```diff
@@ -117,25 +179,46 @@ def _bounds_2d(front: TensorType) -> _BoundedVolumes:
 
         return _BoundedVolumes(lower_result, upper_result)
 
-    @classmethod
-    def _divide_conquer_nd(
-        cls, front: TensorType, jitter: float, threshold: TensorType | float = 0
-    ) -> _BoundedVolumes:
-        # Branch and bound procedure to compute the cells covering the non-dominated region.
-        # Generic version: works for an arbitrary number of objectives.
 
-        len_front, number_of_objectives = front.shape
+class DividedAndConquerNonDominated(BoundIndexPartition):
+    """
+    branch and bound procedure algorithm. a divide and conquer method introduced
+    in :cite:`Couckuyt2012`.
+    """
+
+    def __init__(
+        self, front: TensorType, jitter: float = DEFAULTS.JITTER, threshold: TensorType | float = 0
+    ):
+        """
+        :param front
+        :param jitter
+        :param threshold
+        """
+        tf.debugging.assert_equal(
+            tf.cast(tf.reduce_sum(tf.abs(non_dominated(front)[1])), dtype=front.dtype),
+            tf.zeros(shape=1, dtype=front.dtype),
+            message=f"\ninput {front} " f"contains dominated points",
+        )
+        self.front = tf.gather_nd(front, tf.argsort(front[:, :1], axis=0))  # sort
+        self.front = front
+        self._bounds = self._get_bound_index(jitter, threshold)
+
+    def _get_bound_index(
+        self, jitter: float = DEFAULTS.JITTER, threshold: TensorType | float = 0
```

Comment:
Why is jitter used?

Reply:
it's relating to the working mechanism of divided conqure, in short it always do sth like partition a line into 2 segments and check if the split point is < or > a specified point, so jitter here is used like below to really increase numerical stability.
```
splitted_point = (a-b)/2 # have some numerical inaccracy
if splitted_point + jitter > a_specified_point:
    ...
```

Reply:
Yes, i think we may not need this as arg,  will remove.